src/app/server/AclStorage.cpp

```diff
@@ -0,0 +1,488 @@
+
+struct StagingSubject
+{
+    NodeId nodeId;
+    StagingAuthMode authMode;
+};
+
+CHIP_ERROR Convert(AuthMode from, StagingAuthMode & to)
+{
+    switch (from)
+    {
+    case AuthMode::kPase:
+        to = StagingAuthMode::kPase;
+        break;
+    case AuthMode::kCase:
+        to = StagingAuthMode::kCase;
+        break;
+    case AuthMode::kGroup:
+        to = StagingAuthMode::kGroup;
+        break;
+    default:
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR Convert(StagingAuthMode from, AuthMode & to)
+{
+    switch (from)
+    {
+    case StagingAuthMode::kPase:
+        to = AuthMode::kPase;
+        break;
+    case StagingAuthMode::kCase:
+        to = AuthMode::kCase;
+        break;
+    case StagingAuthMode::kGroup:
+        to = AuthMode::kGroup;
+        break;
+    default:
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR Convert(Privilege from, StagingPrivilege & to)
+{
+    switch (from)
+    {
+    case Privilege::kView:
+        to = StagingPrivilege::kView;
+        break;
+    case Privilege::kProxyView:
+        to = StagingPrivilege::kProxyView;
+        break;
+    case Privilege::kOperate:
+        to = StagingPrivilege::kOperate;
+        break;
+    case Privilege::kManage:
+        to = StagingPrivilege::kManage;
+        break;
+    case Privilege::kAdminister:
+        to = StagingPrivilege::kAdminister;
+        break;
+    default:
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR Convert(StagingPrivilege from, Privilege & to)
+{
+    switch (from)
+    {
+    case StagingPrivilege::kView:
+        to = Privilege::kView;
+        break;
+    case StagingPrivilege::kProxyView:
+        to = Privilege::kProxyView;
+        break;
+    case StagingPrivilege::kOperate:
+        to = Privilege::kOperate;
+        break;
+    case StagingPrivilege::kManage:
+        to = Privilege::kManage;
+        break;
+    case StagingPrivilege::kAdminister:
+        to = Privilege::kAdminister;
+        break;
+    default:
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR Convert(NodeId from, StagingSubject & to)
+{
+    if (IsOperationalNodeId(from) || IsCASEAuthTag(from))
+    {
+        to = { .nodeId = from, .authMode = StagingAuthMode::kCase };
+    }
+    else if (IsGroupId(from))
+    {
+        to = { .nodeId = GroupIdFromNodeId(from), .authMode = StagingAuthMode::kGroup };
+    }
+    else if (IsPAKEKeyId(from))
+    {
+        to = { .nodeId = PAKEKeyIdFromNodeId(from), .authMode = StagingAuthMode::kPase };
+    }
+    else
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR Convert(StagingSubject from, NodeId & to)
+{
+    switch (from.authMode)
+    {
+    case StagingAuthMode::kPase:
+        ReturnErrorCodeIf(from.nodeId & ~kMaskPAKEKeyId, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
```suggestion
        ReturnErrorCodeIf((from.nodeId & ~kMaskPAKEKeyId) != 0, CHIP_ERROR_INVALID_ARGUMENT);
```

Reply:
Added the `!= 0` to all the places using those flags.